spec/controllers/miq_request_controller_spec.rb

```diff
@@ -266,6 +266,52 @@
     end
   end
 
+  context 'showing details of a retirement task' do
+    before do
+      stub_admin
+      EvmSpecHelper.create_guid_miq_server_zone
+    end
+
+    let(:vm1)     { FactoryGirl.create(:vm_cloud) }
+    let(:vm2)     { FactoryGirl.create(:vm_cloud) } # not part of the stack
+    let(:stack)   { FactoryGirl.create(:orchestration_stack).tap { |stack| stack.direct_vms = [vm1] } }
+    let(:service) { FactoryGirl.create(:service_orchestration).tap { |service| service.add_resource!(stack) } }
+    let(:request) { FactoryGirl.create(:miq_service_retirement_request, :options => {:src_ids => [service.id]}) }
+
+    let(:angular_payload) { { :model_name => 'Vm', :parent_id => service.id.to_s } }
+    let(:angular_additional) do
+      {
+        :model             => 'Vm',
+        :parent_id         => service.id.to_s,
+        :parent_class_name => 'ServiceOrchestration',
+        :view_suffix       => 'OrchestrationStackRetireRequest',
+        :display           => 'main',
+        :gtl_type          => 'list'
+      }
+    end
+
+    it 'angular for grid with affected VMs is properly initialized' do
+      payload = angular_payload.update(:report_data_additional_options => angular_additional)
```

Comment:
I'd rather extract this into contexts and then you don't have to call the `update`:
```ruby
let(payload) { {:model => blah, additional_key => additional_val} }
let(:additional_val) { ... }

context 'first' do
  let(:additional_key) { :report_data_additional_options }
  it ...
end

context 'second' do
  let(:additional_key) { :additional_options }
  it ...
end
```

Reply:
That's really smart 👍